### ISSUE TITLE
Some suggestions to your README.md

### ISSUE THREAD

**Issue by sunshineo** [pipeviz/pipeviz#247]
So I spent almost a day on this project.
I fixed your docker build at https://github.com/pipeviz/docker-pipeviz/pull/3
Then I had to do a workaround for
https://github.com/pipeviz/webapp/issues/2
I finally got to this screen. 
![lost](https://cloud.githubusercontent.com/assets/1072766/12380241/02e564c8-bd23-11e5-9272-78695379e18d.jpg)

Then I spent quite some time try to figure out what to do. Eventually I found your wiki pages.https://github.com/pipeviz/pipeviz/wiki/Working%20with%20messages

Could you please add the link on your README?

After read it, it turns out that this is very different from what I was looking for. I have to say you picked a very catch tag line "Your software pipeline, visualized" . Yeah I was trying to visualize my software and the pipline that the data goes through. But your project's focus is clearly on development, test, and deploy pipeline.

I wish you made it clear in your README.md. Again, if it had the link to the wiki, I probably won't look too deep into it.

Good luck with your project!

**Comment by JohnMaguire** [comment]
I thought the README's intro line summarized it quite well:

> Pipeviz is a system for visualizing what you need to know as a software-driven organization: *your development workflow*, the state of your infrastructure, and more. It's a database and visualization webapp rolled into one, fed with data by an ecosystem of agents and other message producers.

From what I understand of this project (I'm just a watcher, I haven't even run this locally yet), its goal is to be able to visualize pretty much whatever you want, provided you feed it data in the correct format.

Infact, I think the following wiki page describes it a lot better than the "Working with messages" page: https://github.com/pipeviz/pipeviz/wiki/Pipeviz-vs-Others

**Comment by JohnMaguire** [comment]
On that note, a clearer mission statement might do the project good: What exactly *is* the goal of pipeviz? Again, I'm not involved in this project other than watching it, but it is hard to grok *exactly* what it's supposed to do for me.

**Comment by sunshineo** [comment]
@JohnMaguire Yeah, I did not read very careful. But the wiki page you provided is not possible to be mistaken.

**Comment by sdboyer** [comment]
Ahh, feedback! Thanks folks :) And extra double-thanks to @sunshineo for investing this effort so far...your perseverance is impressive.

Due to some changeups at work, I'm on a bit of a hiatus from pipeviz at the moment (hence the break in what was otherwise a fairly steady commit stream). That's part of why there aren't better answers here.

But really, you're both entirely right - we need a simple, "here's what this does." Up to now, it's mostly been sufficient for us to give in-person demos (where we can explain as we go), and that's been OK. But more or less our primary focus right now is getting docs and videos (and functionality) together so that it's much easier to understand. But, as @JohnMaguire said:

> its goal is to be able to visualize pretty much whatever you want, provided you feed it data in the correct format.

Yep, pretty much. Have all the little pieces of your infra tell the pipeviz daemon what they know about their corner of the world; pipeviz' engine then reconciles all that together, and the webapp visualizes it.

In that "can visualize anything" sense, pipeviz is similar to D3. But D3 is just the visualization framework: it's on you to decide from where you get the data you're visualizing, how it's all interrelated, and how you deal with changes in the data over time. Pipeviz provides structure to those "where/how/when" questions - it's a full lifecycle solution.

The one main viz we have right now - that one that @sunshineo screenshotted in the OP - is a viz that uses a git commit history (a directed acyclic graph) as a backbone against which you can organize other information with a definable relationship to that graph. Simple example: which of those commits are deployed to 'prod', 'stage', various 'dev' instances, etc.

> After read it, it turns out that this is very different from what I was looking for. I have to say you picked a very catch tag line "Your software pipeline, visualized" . Yeah I was trying to visualize my software and the pipline that the data goes through. But your project's focus is clearly on development, test, and deploy pipeline.

I'm sorry you found the tagline misleading (though I must confess to being happy that you found it catchy :smile:).

Right, sure, that's the case we're focusing on right now, but only because we to start *somewhere*. Looking forward, our aim has been to find more people, more use cases, and flesh out the basic types of information pipeviz knows how to capture and deal with. In the long term, we hope to create a community platform through which data formats and visualization systems could be exchanged - but for now, just working with individuals is the way to go.

So, to that end, I'd love to hear more about what you had in mind for "your software pipeline, visualized."

**Comment by sdboyer** [comment]
Oh, and since y'all found the comparisons to other systems helpful (yay!), I can add a link to that directly from the README.
